### PR TITLE
Introduce v2 settings page

### PR DESCRIPTION
**Pull request by yyorkchan**
ref: [DEV-1844](https://linear.app/authgear/issue/DEV-1844/scaffold-v2-settings-page) [DEV-1845](https://linear.app/authgear/issue/DEV-1845/setting-page-home-show-profile-picture-and-name-of-user)

### CODE REVIEW

File: pkg/lib/config/ui.go [hunk 1]
```diff
@@ -44,6 +48,8 @@ type UIConfig struct {
 	SettingsDisabled       bool `json:"settings_disabled,omitempty"`
 	// Implementation is a temporary flag to switch between authflow and interaction.
 	Implementation UIImplementation `json:"implementation,omitempty"`
+	// SettingImplementation is a temporary flag to switch between setting ui v1 and v2.
+	SettingImplementation SettingUIImplementation `json:"setting_implementation,omitempty"`
```
Comment (cychiuae) [inline pkg/lib/config/ui.go:52]: Setting -> Settings

File: pkg/lib/config/ui.go [hunk 2]
```diff
@@ -100,6 +106,27 @@ func (i UIImplementation) WithDefault() UIImplementation {
...
+	case SettingUIImplementationV2:
+		return i
+	case SettingUIImplementationV1:
+		return i
+	case SettingUIImplementationDefault:
+		fallthrough
+	default:
+		return SettingUIImplementationV1
+	}
+}
```
Comment (cychiuae) [inline pkg/lib/config/ui.go:128]: Setting -> Settings

**Comment by yyorkchan** [comment]
Demo:

https://github.com/user-attachments/assets/a33735da-ba88-40c5-82a9-238c093e1120

Please note that edit profile page is not in the scope of this PR, internal error from this button is expected.

**Comment by yyorkchan** [comment]
This PR is now being referred to [#4640](https://github.com/authgear/authgear-server/pull/4640) from now on.

**Pull request closed without merging**